### ISSUE TITLE
pio_support::pio_die error with `SMS_Ly2_P1x1.1x1_smallvilleIA.ICLM45CNCROP.cori-knl_intel.clm-force_netcdf_pio`

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/E3SM#3122]
Using master of Aug 12th 2019

```
0:  NetCDF: Numeric conversion not representable
0:  pio_support::pio_die:: myrank=          -1 : ERROR: 
0:  pionfwrite_mod::write_nfdarray_double:         250 : 
0:  NetCDF: Numeric conversion not representable
0: Image              PC                Routine            Line        Source             
0: e3sm.exe           00000000228389F1  Unknown               Unknown  Unknown
0: e3sm.exe           0000000020EE6B8A  pio_support_mp_pi         118  pio_support.F90
0: e3sm.exe           0000000020EE4A04  pio_utils_mp_chec          74  pio_utils.F90
0: e3sm.exe           000000002104AF4A  pionfwrite_mod_mp         250  pionfwrite_mod.F90.in
0: e3sm.exe           000000002100B663  piodarray_mp_writ         650  piodarray.F90.in
0: e3sm.exe           000000002100DFB6  piodarray_mp_writ         221  piodarray.F90.in
0: e3sm.exe           0000000021008AD3  piodarray_mp_writ         291  piodarray.F90.in
0: e3sm.exe           0000000020A9E97D  ncdio_pio_mp_ncd_        1799  ncdio_pio.F90.in
0: e3sm.exe           00000000201D26BA  histfilemod_mp_hf        2775  histFileMod.F90
0: e3sm.exe           00000000201C3D6C  histfilemod_mp_hi        3254  histFileMod.F90
0: e3sm.exe           0000000020109550  clm_driver_mp_clm        1438  clm_driver.F90
0: e3sm.exe           00000000200F63DD  lnd_comp_mct_mp_l         509  lnd_comp_mct.F90
0: e3sm.exe           000000002002441F  component_mod_mp_         737  component_mod.F90
0: e3sm.exe           0000000020003EF4  cime_comp_mod_mp_        2602  cime_comp_mod.F90
0: e3sm.exe           000000002002409D  MAIN__                    133  cime_driver.F90
0: e3sm.exe           0000000020001B4E  Unknown               Unknown  Unknown
0: e3sm.exe           000000002293490F  Unknown               Unknown  Unknown
0: e3sm.exe           0000000020001A3A  Unknown               Unknown  Unknown
0: Rank 0 [Mon Aug 12 15:28:13 2019] [c6-1c2s14n3] application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
0: forrtl: error (76): Abort trap signal
0: Image              PC                Routine            Line        Source             
0: e3sm.exe           00000000228443EE  Unknown               Unknown  Unknown
0: e3sm.exe           00000000220DDE00  Unknown               Unknown  Unknown
0: e3sm.exe           0000000022817EE0  Unknown               Unknown  Unknown
0: e3sm.exe           000000002293BE01  Unknown               Unknown  Unknown
0: e3sm.exe           000000002229E472  Unknown               Unknown  Unknown
0: e3sm.exe           000000002233F59B  Unknown               Unknown  Unknown
0: e3sm.exe           00000000222725D5  Unknown               Unknown  Unknown
0: e3sm.exe           0000000020EE6B9D  pio_support_mp_pi         124  pio_support.F90
0: e3sm.exe           0000000020EE4A04  pio_utils_mp_chec          74  pio_utils.F90
0: e3sm.exe           000000002104AF4A  pionfwrite_mod_mp         250  pionfwrite_mod.F90.in
0: e3sm.exe           000000002100B663  piodarray_mp_writ         650  piodarray.F90.in
0: e3sm.exe           000000002100DFB6  piodarray_mp_writ         221  piodarray.F90.in
0: e3sm.exe           0000000021008AD3  piodarray_mp_writ         291  piodarray.F90.in
0: e3sm.exe           0000000020A9E97D  ncdio_pio_mp_ncd_        1799  ncdio_pio.F90.in
0: e3sm.exe           00000000201D26BA  histfilemod_mp_hf        2775  histFileMod.F90
0: e3sm.exe           00000000201C3D6C  histfilemod_mp_hi        3254  histFileMod.F90
0: e3sm.exe           0000000020109550  clm_driver_mp_clm        1438  clm_driver.F90
0: e3sm.exe           00000000200F63DD  lnd_comp_mct_mp_l         509  lnd_comp_mct.F90
0: e3sm.exe           000000002002441F  component_mod_mp_         737  component_mod.F90
0: e3sm.exe           0000000020003EF4  cime_comp_mod_mp_        2602  cime_comp_mod.F90
0: e3sm.exe           000000002002409D  MAIN__                    133  cime_driver.F90
0: e3sm.exe           0000000020001B4E  Unknown               Unknown  Unknown

/global/cscratch1/sd/ndk/acme_scratch/cori-knl/m07-aug12/SMS_Ly2_P1x1.1x1_smallvilleIA.ICLM45CNCROP.cori-knl_intel.clm-force_netcdf_pio.r00
```

**Comment by ndkeen** [comment]
Trying again with DEBUG, I see a different (likely different) error:

```
0: forrtl: severe (408): fort: (2): Subscript #1 of the array FORC_RAIN has value 4 which is greater than the upper bound of 1
0:
0: Image              PC                Routine            Line        Source
0: e3sm.exe           000000002616E1F0  Unknown               Unknown  Unknown
0: e3sm.exe           0000000022AEE8FF  phenologymod_mp_c        2267  PhenologyMod.F90
0: e3sm.exe           0000000022A4B50D  phenologymod_mp_p         260  PhenologyMod.F90
0: e3sm.exe           0000000022758030  ecosystemdynmod_m         627  EcosystemDynMod.F90
0: e3sm.exe           00000000204C669E  clm_driver_mp_clm        1029  clm_driver.F90
0: e3sm.exe           000000002046E9DA  lnd_comp_mct_mp_l         509  lnd_comp_mct.F90
0: e3sm.exe           000000002005DABB  component_mod_mp_         737  component_mod.F90
0: e3sm.exe           00000000200274F9  cime_comp_mod_mp_        2602  cime_comp_mod.F90
0: e3sm.exe           0000000020045C08  MAIN__                    133  cime_driver.F90
0: e3sm.exe           0000000020001B4E  Unknown               Unknown  Unknown
0: e3sm.exe           000000002624F20F  Unknown               Unknown  Unknown
0: e3sm.exe           0000000020001A3A  Unknown               Unknown  Unknown
```

**Comment by bbye** [comment]
This is caused by issue #3110  and is fixed with PR #3111 . I confirmed that this test now passes on cori with the bug fix PR.
